src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -580,14 +579,9 @@ Real SpatialPooler::initPermNonConnected_() {
 }
 
 
-vector<Real> SpatialPooler::initPermanence_(const vector<UInt> &potential, //TODO make potential sparse
-                                            Real connectedPct) {
+const vector<Real> SpatialPooler::initPermanence_(const SDR &potential, const Real connectedPct) {
   vector<Real> perm(numInputs_, 0);
-  for (UInt i = 0; i < numInputs_; i++) {
-    if (potential[i] < 1) {
-      continue;
-    }
-
+  for (const auto i : potential.getSparse()) { //TODO add SDR iterator over sparse indices?
```

Comment:
I don't understand this TODO note.  Doesn't `i` iterator over the sparse indices?

Reply:
I mean `for idx : SDR`, but I'll rm the note, it's just more code for a trivially accessible functionality